hudi-common/src/main/java/org/apache/hudi/common/model/HoodieIndexDefinition.java

```diff
@@ -45,14 +48,14 @@ public class HoodieFunctionalIndexDefinition implements Serializable {
   // Any other configuration or properties specific to the index
   private Map<String, String> indexOptions;
 
-  public HoodieFunctionalIndexDefinition() {
+  public HoodieIndexDefinition() {
   }
 
-  public HoodieFunctionalIndexDefinition(String indexName, String indexType, String indexFunction, List<String> sourceFields,
-                                         Map<String, String> indexOptions) {
+  public HoodieIndexDefinition(String indexName, String indexType, String indexFunction, List<String> sourceFields,
```

Comment:
So after this change, we only have secondary index and the notion of functional index is removed.